libs/agno/agno/agent/agent.py

```diff
@@ -688,6 +689,65 @@ def set_tools(self, tools: List[Union[Toolkit, Callable, Function, Dict]]):
         self.tools = tools
         self._rebuild_tools = True
 
+    def _initialize_session_state(self, user_id: Optional[str] = None, session_id: Optional[str] = None) -> None:
+        self.session_state = self.session_state or {}
+        if user_id is not None:
+            self.session_state["current_user_id"] = user_id
+            if self.team_session_state is not None:
+                self.team_session_state["current_user_id"] = user_id
```

Comment:
Thought: maybe instead of carrying this team_session_state in the agents, we can store it only in the team, and access it always through it (`agent.team.session_state`)

Just thinking that in cases like this (updating the user_id of the team_session_state), this could happen only once if we do it in the team and not per member